packages/ui/Slottable/Slottable.js

```diff
@@ -103,16 +43,32 @@ const defaultConfig = {
  */
 const Slottable = hoc(defaultConfig, (config, Wrapped) => {
 	const slots = config.slots;
-	return kind({
-		name: 'Slottable',
-		render: (props) => {
-			const adjusted = distribute(slots, props);
-			return (
-				<Wrapped {...adjusted} />
-			);
-		}
-	});
+
+	// eslint-disable-next-line no-shadow
+	return function Slottable ({children, ...rest}) {
+		const distributed = useSlots({slots, children});
+
+		// Slottable allows there to be other values in the destination slot and merges them.
+		// However, consumers can't avoid key warnings when merging the two lists so we should
+		// "consider this harmful" and not continue to support this with the hook and instead
+		// encourage "either/or". We should continue to support it here so this block ensures
+		// backwards compatibility.
+		slots.forEach(slot => {
+			const dist = distributed[slot];
+			const prop = rest[slot];
+			if (prop && dist) {
+				distributed[slot] = [].concat(prop, dist);
```

Comment:
This is weird. I don't think I like it. I didn't even realize the old version had this behavior, and I'm probably the most frequent user of Slots. Mostly don't like it because A) I don't know that it's documented anywhere B) you can't control the order, so if it doesn't accidentally do what you want perfectly, you're SOL.

I feel like it's totally acceptable for Slottable (useSlots) to simply decide one of the two is "more valid" than the other, the slot or the prop and always prefer that one, provided it's consistent, predictable, overridable, and documented. I thin the following patterns could support those requirements with intuitive DX:

```jsx
<Component
   title="My Title"
   bodyText="potatoes are tasty"
>
   <bodyText>Only with salt on them</bodyText>
   <buttons><button>A Button!</button></buttons>
</Component>
```
The props argument can accept "fallback" values for if child-slots are not provided by the end-developer, in this example we have only props with names mapping 1:1 with existing slots.
```javascript
{children, title, bodyText} = props;
slots = ['title', 'bodyText', 'buttons'];
distributed = useSlots({slots, {children, title, bodyText}); // didn't include "buttons" prop, don't care about `buttons` prop in this case

// where distributed ==
{
   // title is fed back as a prop, so it's indistinguishable from whether it came in as a prop or a child-slot
   title: 'My Title',

   // bodyText child-slot is preferred since it seems more intentional, given that if the component author had wanted this to just be a prop, it wouldn't have been one, yet can be supported if a child-slot isn't provided, for convenience
   bodyText: 'Only with salt on them',

   // buttons comes in as a child-slot and we return it into the distributed collection. The author expects no comparison or override, since it wasn't intentionally included in the useSlots props argument.
   buttons: <button>A Button!</button>
}
```

Reply:
Agreed! I'll make an update and we can review it again.